### PR TITLE
Update go version

### PR DESCRIPTION
**Pull request by sivchari**
**What this PR does / why we need it**:

Update Go version to 1.20.3.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/server/grpcapi/api.go [hunk 1]
```diff
@@ -343,6 +344,46 @@ func (a *API) ListApplications(ctx context.Context, req *apiservice.ListApplicat
 	}, nil
 }
 
+func (a *API) UpdateApplication(ctx context.Context, req *apiservice.UpdateApplicationRequest) (*apiservice.UpdateApplicationResponse, error) {
```
Comment (khanhtc1202) [inline pkg/app/server/grpcapi/api.go:347]: this change does not related to this pr, right? 👀

Comment (sivchari) [reply pkg/app/server/grpcapi/api.go:347]: Oops, I'll fix it 🙈

**Comment by khanhtc1202** [comment]
Thanks! Could you also update go version for pipecd tools as well (you can find it under `tool/` directory) 😄

**Review by khanhtc1202 — APPROVED** [review]
Nice, thank you 🙌

**Pull request merged**